agent/consul/leader.go

```diff
@@ -126,9 +129,23 @@ func (s *Server) monitorLeadership() {
 	}
 }
 
+func (s *Server) leadershipTransfer() {
+	retryCount := 3
+	for i := 0; i < retryCount; i++ {
```

Comment:
Should this function take some action if the leadershipTransfer is unsuccessful after the 3 tries?

Reply:
Maybe the `weAreLeaderCh` should be closed and set to nil, since if we then receive from `aclUpgradeCh` and that channel is still open, we initialize ACLs when we probably shouldn't.

Reply:
I changed my PR a bit, `leaderLoop` is only left if `transferLeadership` was successful. I think that answers your question, because `leaderLoop` will retry `transferLeadership` if it failed before.